test/Fields/field_opt.jl

```diff
@@ -51,3 +52,26 @@ include(joinpath(@__DIR__, "util_spaces.jl"))
         @test p == 0
     end
 end
+
+# https://github.com/CliMA/ClimaCore.jl/issues/949
+@testset "Allocations with getproperty on FieldVectors" begin
+    FT = Float64
+    function allocs_test!(Y)
+        x = Y.x
```

Comment:
This could be getting optimized away a no-op. You may have to do something with `x`
```suggestion
        x = Y.x
        fill!(x, 2.0)
```